app/src/main/java/com/tananaev/logcat/LineAdapter.java

```diff
@@ -99,8 +146,38 @@ public LineViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     @Override
     public void onBindViewHolder(LineViewHolder holder, int position) {
         Line item = linesFiltered.get(position);
-        holder.getTextView().setText(item.getContent());
+        holder.itemView.setTag(item);
+        holder.itemView.setOnLongClickListener(onItemLongClickListener);
+        String text = item.getContent();
+        if (!TextUtils.isEmpty(lowerKeyword) || !TextUtils.isEmpty(lowerSearchWord)) {
+            SpannableString spannableText = new SpannableString(text);
+            String lowerContent = item.getLowerContent();
+
+            // filter keyword
+            if (!TextUtils.isEmpty(lowerKeyword)) {
+
+                int index = 0, found;
+                while ((found = lowerContent.indexOf(lowerKeyword, index)) >= 0) {
+                    spannableText.setSpan(new BackgroundColorSpan(Color.RED), found, found + lowerKeyword.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+                    index = found + lowerKeyword.length();
+                }
+            }
+
+            // highlight search word
```

Comment:
I don't think there is any value in this comment. It's pretty obvious what we are doing here.